secboot/secboot_tpm.go

```diff
@@ -273,9 +276,73 @@ func UnlockVolumeUsingSealedKeyIfEncrypted(disk disks.Disk, name string, sealedE
 	}
 }
 
+// FDERevealKeyRequest carries the operation and parameters for the
+// fde-reveal-key binary to support unsealing keys that were sealed
+// with the "fde-setup" hook.
+type FDERevealKeyRequest struct {
+	Op string `json:"op"`
+
+	SealedKey     []byte `json:"sealed-key"`
+	SealedKeyName string `json:"sealed-key-name"`
+
+	// TODO: add VolumeName,SourceDevicePath later
+}
+
+// fdeRevealKeyRuntimeMax is the maximum runtime a fde-reveal-key can execute
+// XXX: what is a reasonable default here?
+var fdeRevealKeyRuntimeMax = "2m"
+
+// fdeRevealKeyCommand returns a *exec.Cmd that is suitable to run
+// fde-reveal-key using systemd-run
+func fdeRevealKeyCommand() *exec.Cmd {
+	return exec.Command(
+		"systemd-run",
+		"--pipe", "--same-dir", "--wait", "--collect",
+		"--service-type=exec",
+		"--quiet",
+		// ensure we get some result from the hook within a
+		// reasonable timeout and output from systemd if
+		// things go wrong
+		fmt.Sprintf("--property=RuntimeMaxSec=%s", fdeRevealKeyRuntimeMax),
+		`--property=ExecStopPost=/bin/sh -c 'if [ "$EXIT_STATUS" != 0 ]; then echo "service result: $SERVICE_RESULT" 1>&2; fi'`,
```

Comment:
can we test this bit by mocking fde-reveal-key and not systemd-run itself or more precisely mocking the latter just in the sense of adding --user?

Reply:
I updated this, I have two tests now one that mocks the entire systemd-run and one that just mocks fde-reveal-key, probably too much to have both. Also need to do the failure case still :/

Reply:
I updated the code now to test the happy path with systemd-run --user and also the unhappy case output.